src/main/scala/com/atomist/rug/kind/scala/ScalaMetaTreeBackedTreeNode.scala

```diff
@@ -0,0 +1,35 @@
+package com.atomist.rug.kind.scala
+
+import com.atomist.tree.TreeNode
+import com.atomist.tree.content.text.{InputPosition, OffsetInputPosition, PositionedTreeNode}
+
+import scala.meta._
+
+/**
+  * Immutable tree backed by a ScalaMeta Tree
+  */
+private[scala] class ScalaMetaTreeBackedTreeNode(smTree: Tree)
+  extends TreeNode with PositionedTreeNode {
+
+  override val nodeName: String = {
+    val fqn = smTree.getClass.getName
+    fqn.drop(fqn.lastIndexOf("$") + 1).replace("Impl", "")
```

Comment:
Good point. I'll put a comment in after I merge, about what the class names look like before and after.